### PR TITLE
move to C++14

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

**Comment by mgeplf** [comment]
@NadirRoGue would like to use some C++14 features in a PR; so far we haven't needed it, but we may as well move.

Please add others who may be interested.

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,7 +44,7 @@ option(${PROJECT_NAME}_CXX_WARNINGS "Compile C++ with warnings as errors" ON)
 set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/CMake)
 set(CMAKE_SKIP_INSTALL_ALL_DEPENDENCY true)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
```
[the remark this answers is not in the document]
Comment (mgeplf) [reply CMakeLists.txt:47]: Small steps, perhaps that will come later.

[the remark this answers is not in the document]
Comment (NadirRoGue) [reply CMakeLists.txt:47]: I believe the only constraint is that all office workstations should support it natively, and there are still people on Ubuntu 18

[the remark this answers is not in the document]
Comment (matz-e) [reply CMakeLists.txt:47]: Really?

**Review by matz-e — APPROVED** [review]
Sigh… better than nothing. Only trailing 7 years past current times.

**Review by NadirRoGue — APPROVED** [review]

**Pull request merged**